File: LibraryAPI/LibraryAPI/Entities/Book.cs
namespace LibraryAPI.Entities;

public class Book
{
    public Guid Id { get; set; }
    public required string Title { get; set; }
    public required string Author { get; set; }
    public DateOnly PublishDate { get; set; }
    public string? Isbn { get; set; }
    public int Quantity { get; set; }
    public int AvailableQuantity { get; set; }
    public string? Description { get; set; }
    
    public Guid CategoryId { get; set; }
    public Category? Category { get; set; }
    
    public ICollection<BookBorrowingRequestDetails> BookBorrowingRequestDetails { get; set; }
}

File: web-app/src/types/Book.ts
export interface Book {
    id: string
    title: string
    author: string
    categoryId: string
    availableQuantity: number,
    quantity: number
    isbn: string
    publishDate: string
    description?: string
}

export interface BookFormData {
    title: string
    author: string
    categoryId: string
    availableQuantity: number
    quantity: number
    isbn: string
    publishDate: string
    description?: string
}

export interface BorrowingRequest {
    id: string
    userId: string
    bookId: string
    requestDate: string
    status: number
    processedDate?: string
    dueDate?: string
    processedBy?: string
}

export interface User {
    id: string
    name: string
    email: string
    phoneNumber: string
}
export interface BookBorrowing {
    id: string;
    bookId: string;
    bookTitle: string;
}
export interface BorrowingResponse {
    id: string
    requestDate: string
    status: number
    expirationDate: string
    requestor: User
    books: BookBorrowing[];
}